lib/bucket_store/key_storage.rb

```diff
@@ -40,45 +124,31 @@ def filename
     # @example Download a key
     #   BucketStore.for("inmemory://bucket/file.xml").download
     def download
-      raise ArgumentError, "Key cannot be empty" if key.empty?
-
-      BucketStore.logger.info(event: "key_storage.download_started")
-
-      start = BucketStore::Timing.monotonic_now
-      result = adapter.download(bucket: bucket, key: key)
-
-      BucketStore.logger.info(event: "key_storage.download_finished",
-                              duration: BucketStore::Timing.monotonic_now - start)
-
-      result
+      buffer = StringIO.new
+      stream.download(file: buffer).
+        except(:file).
+        merge(content: buffer.string)
```

Comment:
Not that happy that we're creating a couple of hashes here. Imagine memory usage of this is marginal but might end up with more GC pressure

Reply:
well, fun fact: `except` isn't available in Ruby 2.7 which this needs to support so going with a more memory safe option of altering the hash inline